### ISSUE TITLE
CIMDateTime does not enforce datetime awareness

### ISSUE THREAD

**Issue by KSchopmeyer** [pywbem/pywbem#698]
The definition of the dtarg in CIMDateTIme includes the following in its documentation:

```          
 * A :class:`py:datetime.datetime` object must be timezone-aware
              (see :class:`~pywbem.MinutesFromUTC`) and will result in a point
              in time.
```
However the code does not include a test to insure awareness (i.e. existence of tzinfo) although the documentation says that the tzone must be part of the constructor. We have bunch of examples in our tests where the tzone is not there.

One issue is that if you use the Pywbem CIMDateTime compare function to compare naive and aware date times it generates a TypeError exception rather than returning a False indicating that they are not equal.

The other issue is that the documentation and code to not match.

Found this a recorder test where the datetime naive definition input with the request was compared with a datetime aware version produced by yaml dump/load which results in a TypeError exception: niave and aware datetimes cannot be compared)

It would be a real incompatibility to force the change to insure that tzinfo was part of the interface but this leaves us open to the TypeError exception when simply comparing two CIMDateTime objects.

**Comment by andy-maier** [comment]
The CIM datetime type is clearly always timezone-aware, i.e. it stands on its own without external knowledge about an implied timezone. So if we allow a timezone-naive Python datetime object as input to the `CIMDateTime` init routine, we have to define an implied timezone for that case. If I look at the behavior of `CIMDateTime`, e.g. in its conversion to a CIM datetime string, then it does assume UTC as a default timezone for that case.

So as an alternative, we could leave the code as is and change the description to state that timezone-native datetime input is interprteed as being in UTC. Note that this default is not quite consistent with the default of local timezone we state for the `tzi` argument of `CIMDateTime.now()` and `.fromtimestamp()`, but then the meaning of that argument is different. It would still be an inconsistency.

**Comment by KSchopmeyer** [comment]
We want to keep compatible behavior.

The most logical is to change the code so that we set the timezone to utc if it does not exist in the constructor.

However, since the this code is messy anyway, it would be logical to make this change as part of a more general cleanup of this code.

**Comment by KSchopmeyer** [comment]
We agreed that we to change a niave definition to be an aware definition in utc.

**Comment by andy-maier** [comment]
Fixed with PR #715.
